lib/record_store/version.rb

```diff
@@ -1,3 +1,3 @@
 module RecordStore
-  VERSION = '5.7.0'.freeze
+  VERSION = '5.8.0'.freeze
```

Comment:
Would probably go for `5.7.1` instead as this is not really a new feature if we don't change behavior nor break it.
It's just transparent for the user, so a 'patch' makes sense to me here (we're patching the behavior to fit with what should be done and avoid rate limits). Maybe @cttttt will have a different feeling though?

Reply:
I see. Yeah I was debating between 'new feature' and 'patch'. I decided to go with 'feature' because I believe it change bahavior (now it controls the rate limit by 'sleep'). But I also get your point. I want to wait for @cttttt 's thought as well.

Reply:
My thought generally is:
- Does it remove any existing function or change the way to call any existing function ? => major
- Does it add a new function or new parameters to an existing function without breaking its current behavior for users ? => minor
- Else => patch (basically, does it do anything in the background that the user won't be impacted by other than positively, like fixing bugs or adding a process to avoid bugs)

But I'm not really caring about the version number. I just prefer to avoid reaching version 5.567.0 too soon :p

Reply:
As long as this is well tested (and practically, as long as it touches this provider only (it hasn't really seen prime time)) a patch release sounds good 👍🏽.

@XaF 's impact `switch` is pretty spot on.  This update isn't intended to change behaviour so it should be okay if folks automatically pick it up.  See https://semver.org/ for the canonical meanings of major, minor, and patch.

Reply:
Cool, now it's solid that we are going with patch, `5.7.1`.